TelemetryService/CustomPIIFilter.cs

```diff
@@ -69,17 +70,15 @@ public class CustomPIIFilter : ITelemetryProcessor
         private const string ODataFilterOperator = "$filter";
         private const string SecurityMask = "****";
 
-        public CustomPIIFilter(ITelemetryProcessor next, UriTemplateMatcher uriTemplateMatcher)
+        public CustomPIIFilter(ITelemetryProcessor next, IServiceProvider serviceProvider)
         {
             _next = next
                 ?? throw new ArgumentNullException(nameof(next), $"{ next }: { nameof(next) }");
-
-            if (uriTemplateMatcher == null)
+            if (serviceProvider == null)
             {
-                throw new ArgumentNullException(nameof(uriTemplateMatcher), $"{ uriTemplateMatcher }: { nameof(uriTemplateMatcher) }");
+                throw new ArgumentNullException(nameof(serviceProvider), $"{ serviceProvider }: { nameof(serviceProvider) }");
             }
-
-            _uriTemplateMatcher = uriTemplateMatcher;
+            _serviceProvider = serviceProvider;
```

Comment:
```suggestion
            _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider), $"{ serviceProvider }: { nameof(serviceProvider) }");;
```

And delete the block above